src/modules/bonus/components/PayWithBonusPointsCheckbox.tsx

```diff
@@ -57,12 +59,20 @@ export const PayWithBonusPointsCheckbox = ({
       ),
     );
 
+  const onPressWithLogging = () => {
+    onPress();
+    logEvent('Bonus', 'bonus points checkbox toggled', {
+      bonusProductId: bonusProduct.id,
+      newState: isChecked,
+    });
```

Comment:
Would it make more sense to put this directly inside the `onPress`? the other change in this commit has the functions directly inside it. Just thinking out loud here 🤔